examples/sampling/graphbolt/temporal_link_prediction.py

```diff
@@ -0,0 +1,367 @@
+"""
+This script trains and tests a Heterogeneous GraphSAGE model for link
+prediction with temporal information using graphbolt dataloader.
+
+Unlike previous dgl examples, we've utilized the newly defined dataloader
+from GraphBolt. This example will help you grasp how to build an end-to-end
+training pipeline using GraphBolt.
+
+While node classification predicts labels for nodes based on their
+local neighborhoods, link prediction assesses the likelihood of an edge
+existing between two nodes, necessitating different sampling strategies
+that account for pairs of nodes and their joint neighborhoods.
```

Comment:
Unrelated, can be removed, maybe talk more about how what's temporal in the comment.